c/include/brotli/port.h

```diff
@@ -47,7 +55,7 @@
 #define BROTLI_MODERN_COMPILER 0
 #endif
 
-#if defined(BROTLI_SHARED_COMPILATION) && defined(_WIN32)
+#if defined(BROTLI_SHARED_COMPILATION) && (__has_declspec_attribute(dllexport) && __has_declspec_attribute(dllimport))
```

Comment:
Also, for more uniform code, it is better to use semantic derivative of check - i.e. define
BROTLI_DLLIMPORT and BROTLI_DLLEXPORT (like here http://llvm.org/viewvc/llvm-project/cfe/trunk/docs/LanguageExtensions.rst?view=diff&r1=223466&r2=223467&pathrev=223467) and use them in place of __declspec(xxx)